js/foam/u2/SearchBorder.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * @license
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+CLASS({
+  package: 'foam.u2',
+  name: 'SearchBorder',
+  extends: 'foam.u2.View',
+  documentation: 'Wraps a view of a DAO to add filtering based on a search ' +
+      'query. You can supply a $$DOC{ref:".queryParser"}, or use the default ' +
+      '(KEYWORD).',
+
+  imports: [
+    'data',
+  ],
+  exports: [
+    'filteredProxy_ as data',
+  ],
+
+  properties: [
+    {
+      name: 'data',
+    },
+    {
+      name: 'queryParser',
+      documentation: 'A query parser, or a function from string to EXPR.',
+      defaultValue: function(s) { return KEYWORD(s); }
+    },
+    {
+      name: 'delegate',
+      documentation: 'An inner view that expects a DAO. Binds it to filtered_.',
+      postSet: function(old, nu) {
+        nu.data = this.filtered_$Proxy;
+      },
+    },
+    {
+      type: 'String',
+      name: 'search',
+      toPropertyE: function(X) {
+        return X.lookup('foam.u2.Input').create({
+          inline: true,
+          onKeyMode: true
```

Comment:
I'm aware. It isn't actually binding properly at all. That isn't a regression of this PR; it's broken in the U2 browser generally. There are a couple of other problems I've got written down, too.